### PR TITLE
[vcpkg-cmake] Fix the application of VCPKG_CMAKE_CONFIGURE_OPTIONS with more than one option

### PR DESCRIPTION
**Pull request by TheJCAB**
**Describe the pull request**

VCPKG_CMAKE_CONFIGURE_OPTIONS (plural) is meant to be a list of options to be added to a cmake invocation.

But it was being applied quoted, which disallowed it from containing multiple options.
- If it was initialized with a list, cmake would get the options ";"-separated.
- If it was initialized with a blank-separated string, cmake would get the entire string quoted as a single argument.

The fix is simple: don't (ever) quote arguments to list(APPEND) unless they are really meant to be a single element to append to the list.

- #### What does your PR fix?

Fixes an issue I encountered when adding a custom triplet that uses a toolchain file.

I see issues #17750 and #23741 (and 23852 more indirectly) all contain reports of multiple options working correctly. Apparently PR #23259 inadvertently broke it by substituting list() with vcpkg_list().

vcpkg_list() top comments say "A replacement for CMake's `list()` function, which correctly handles elements
with internal semicolons (in other words, escaped semicolons)", but fail to explain what's the nature of the issue, and the code is  not trivial to follow, so I'm guiding myself from empirical observation. Explanations and insights are welcome. In the meantime, removing the quotes here works around the issue.

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?

N/A

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?

Yes (it's just basically a typo fix).

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?

N/A

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

**Comment by TheJCAB** [comment]
> _This is a new experimental fast check for PR issues. Please let us know if this bot is helpful!_
> 
> PRs must add only one version and must not modify any published versions
> When making any changes to a library, the version or port-version in `vcpkg.json` or `CONTROL` must be modified.

I don't understand what the bot is ask for, here. This seems unfriendly to the uninitiated. I'm just fixing basically a typo-like issue :-).

**Comment by TheJCAB** [comment]
> I don't understand what the bot is asking for, here.

@walbourn assisted offline. Appreciated.

**Comment by TheJCAB** [comment]
And after modifying the version, now it all has a merge conflict with another PR that beat me by a matter of minutes.

I am sending this PR from a fork (github way of doing things?). I don't know how to pull the new master branch into my fork so I can re-resolve the conflict (git fetch does nothing), and I don't know what the correct resolution is now:

- Just increase version-date.
- Just increase port-version.
- Increase both.

More assistance appreciated.

**Comment by JackBoosY** [comment]
I think this changes is correct, @Neumann-A what do you think about?

**Comment by dg0yt** [comment]
> vcpkg_list() top comments say "A replacement for CMake's `list()` function, which correctly handles elements
> with internal semicolons (in other words, escaped semicolons)", but fail to explain what's the nature of the issue, and the code is not trivial to follow, so I'm guiding myself from empirical observation. Explanations and insights are welcome. 

`vcpkg_list()` is one of the few functions which has a unit test. This can give insight in some of the quirks resolved by the function:
https://github.com/microsoft/vcpkg/blob/master/scripts/test_ports/unit-test-cmake/test-vcpkg_list.cmake

> In the meantime, removing the quotes here works around the issue.

I think this is the right change here.

**Comment by JackBoosY** [comment]
There is one thing that need to be noticed is that this issue may also happened in other functions `vcpkg_configure_*`.

**Review by Neumann-A — APPROVED** [review]

**Comment by Neumann-A** [comment]
`VCPKG_CMAKE_CONFIGURE_OPTIONS` is meant to splash onto `vcpkg_list` since `vcpkg_list` handles all elements correctly.
If it would have been just `list()` the extra quotes would have been necessary.

**Review by JackBoosY — APPROVED** [review]

**Comment by TheJCAB** [comment]
I guess I don't have write access to merge this? Appreciate it if someone else does it before conflicts creep in.

File: versions/v-/vcpkg-cmake.json [hunk 1]
```diff
@@ -1,9 +1,9 @@
 {
   "versions": [
     {
-      "git-tree": "8b07d914c90cf8f611973318c85d3af13201e3f9",
-      "version-date": "2022-05-10",
-      "port-version": 1
+      "git-tree": "819e45a14fb875ec7e8373143c994b7bd8d8f7cb",
+      "version-date": "2022-06-07",
+      "port-version": 0
```
Comment (Thomas1664) [inline versions/v-/vcpkg-cmake.json:6]: Please don't overwrite existing versions.

Comment (JackBoosY) [reply versions/v-/vcpkg-cmake.json:6]: Ops sorry, I ignor this.

Comment (JackBoosY) [reply versions/v-/vcpkg-cmake.json:6]: Please restore this file, rerun command `x-add-version vcpkg-cmake` then commit changes.

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: versions/v-/vcpkg-cmake.json (hunk 1 above)
Comment (TheJCAB) [reply versions/v-/vcpkg-cmake.json:6]: Oh... right. My bad. I plead noobness.

**Review by JackBoosY — APPROVED** [review]

**Pull request merged**